crates/iota-types/src/iota_system_state/iota_system_state_inner_v2.rs

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+
+use super::{
+    AdvanceEpochParams, IotaSystemStateTrait,
+    epoch_start_iota_system_state::EpochStartValidatorInfoV1,
+    get_validators_from_table_vec,
+    iota_system_state_inner_v1::{StorageFundV1, SystemParametersV1, ValidatorSetV1, ValidatorV1},
+    iota_system_state_summary::{IotaSystemStateSummary, IotaValidatorSummary},
+};
+use crate::{
+    balance::Balance,
+    base_types::IotaAddress,
+    collection_types::{Bag, Table, TableVec, VecMap, VecSet},
+    committee::{CommitteeWithNetworkMetadata, NetworkMetadata},
+    error::IotaError,
+    gas_coin::IotaTreasuryCap,
+    iota_system_state::epoch_start_iota_system_state::EpochStartSystemState,
+    storage::ObjectStore,
+    system_admin_cap::IotaSystemAdminCap,
+};
+
+/// Rust version of the Move iota_system::iota_system::IotaSystemStateV1 type
+/// An additional field `safe_mode_computation_charges_burned` is added over
+/// IotaSystemStateV1 to allow for burning of base fees in safe mode when
+/// protocol_defined_base_fee is enabled in the protocol config.
```

Comment:
Nit: Use implied reference links where possible. E.g.
```suggestion
/// Rust version of the Move [`iota_system::iota_system::IotaSystemStateV1`] type
/// An additional field `safe_mode_computation_charges_burned` is added over
/// [`IotaSystemStateV1`](super::iota_system_state_inner_v1::IotaSystemStateV1) to allow for burning of base fees in safe mode when
/// protocol_defined_base_fee is enabled in the protocol config.
```
Also have a look at common conventions for Rust API docs: https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md#appendix-a-full-conventions-text